### PR TITLE
feat: New Queries for APIM

### PR DESCRIPTION
**Pull request by jacopocarlini**
#### List of Changes

- added queries for Apim
- added `service` name as config parameter

The tool uses different queries if you set as `data_source` a resource of type Microsoft.ApiManagement

#### Motivation and Context

To use the tool for services not exposed on internet.

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/opex_dashboard/templates/azure_dashboard_terraform.tf [hunk 1]
```diff
@@ -35,9 +35,15 @@ resource "azurerm_monitor_scheduled_query_rules_alert" "alarm_availability_{{ fo
   enabled                 = true
   auto_mitigation_enabled = false
 
+  {% if resource_type == 'app-gateway' %}
   query = <<-QUERY
-    {% include "queries/availability.kusto" with is_alarm=True threshold=props.availability_threshold %}
+    {% include "gateway_queries/availability.kusto" with is_alarm=True threshold=props.availability_threshold %}
   QUERY
+  {% else %}
```
Comment (pasqualedevita) [inline src/opex_dashboard/templates/azure_dashboard_terraform.tf:42]:
```suggestion
  {% if resource_type == 'api-management' %}
```

Comment (pasqualedevita) [reply src/opex_dashboard/templates/azure_dashboard_terraform.tf:42]: Please add more strict checks :)

File: src/opex_dashboard/templates/azure_dashboard_terraform.tf [hunk 2]
```diff
@@ -35,9 +35,15 @@ resource "azurerm_monitor_scheduled_query_rules_alert" "alarm_availability_{{ fo
...
+  query = <<-QUERY
+    {% include "apim_queries/availability.kusto" with is_alarm=True threshold=props.availability_threshold %}
```
Comment (pasqualedevita) [inline src/opex_dashboard/templates/azure_dashboard_terraform.tf:44]:
```suggestion
    {% include "azure/api-management_queries/availability.kusto" with is_alarm=True threshold=props.availability_threshold %}
```
Can we rename the folder as the resource-type parameter? In this way we can have a dynamic tool :)

- azure
- - api-management
- - app-gateway

```
    {% include "azure/<<props.resource-type>>_queries/availability.kusto" with is_alarm=True threshold=props.availability_threshold %}
```

**Review by pasqualedevita — APPROVED** [review]

**Review by pasqualedevita — CHANGES_REQUESTED** [review]
see comments

File: src/opex_dashboard/templates/azure_dashboard_terraform.tf (hunk 2 above)
Comment (jacopocarlini) [reply src/opex_dashboard/templates/azure_dashboard_terraform.tf]: Done

**Review by pasqualedevita — APPROVED** [review]
Top @jacopocarlini <3

**Pull request merged**